torch_geometric/transforms/add_metapaths.py

```diff
@@ -145,3 +155,20 @@ def __call__(self, data: HeteroData) -> HeteroData:
                     del data[node]
 
         return data
+
+    def _edge_index_sampling(self, edge_index, max_sample, backward=False):
```

Comment:
```suggestion
    def _edge_index_sampling(self, adj, max_sample, backward=False):
```
Let's work on the `SparseTensor` here.